### PR TITLE
Allow for jwtAuth being provided inside of F[_]

### PR DESCRIPTION
**Pull request by froth**
This allows for the keys being stored in a database or fetched from an api.

I opted for failing the request if the IO raises an error instead of responding with Forbidden. I believe, that this would be more in the range of InternalServerError for most applications but can be decided by whoever uses the library.

### CODE REVIEW

File: core/src/main/scala/dev/profunktor/auth/middleware.scala [hunk 1]
```diff
@@ -14,6 +14,12 @@ object JwtAuthMiddleware {
   def apply[F[_]: MonadThrow, A](
       jwtAuth: JwtAuth,
       authenticate: JwtToken => JwtClaim => F[Option[A]]
+  ): AuthMiddleware[F, A] =
+    apply(Monad[F].pure(jwtAuth), authenticate)
```
Comment (gvolpe) [inline core/src/main/scala/dev/profunktor/auth/middleware.scala]: Nitpick: Could use `jwtAuth.pure[F]` here instead? You can remove the `Monad` import.

**Review by gvolpe — CHANGES_REQUESTED** [review]
Thank you! Looks good, just a minor comment.

File: core/src/main/scala/dev/profunktor/auth/middleware.scala (hunk 1 above)
Comment (froth) [reply core/src/main/scala/dev/profunktor/auth/middleware.scala]: done

**Comment by froth** [comment]
Thanks for the fast reply. I implemented your suggestion.

**Review by gvolpe — APPROVED** [review]
LGTM 👍🏽

**Comment by froth** [comment]
Would it be possible to get a minor release for this? Maybe with updated library dependencies? I can open an PR with dependency updates if this helps.

**Comment by gvolpe** [comment]
There will be a SNAPSHOT published as soon as this is merged. But yes, a proper release should come up shortly.

**Comment by froth** [comment]
Sounds good :)

**Pull request merged**